### PR TITLE
release/2.0.0 - Make DBNull serializable and adding tests

### PR DESCRIPTION
**Pull request by ViktorHofer**
Requires: https://github.com/dotnet/coreclr/pull/13906
Issue: dotnet/corefx#23398
Docs update: https://github.com/dotnet/docs/pull/3116

**Blobs need to be updated after coreclr change is merged.**

@weshaggard I'm disabling the typeforward in the release/2.0.0 branch for uap/uapaot.

waiting for shiproom approval.

### CODE REVIEW

File: src/shims/manual/mscorlib.forwards.cs [hunk 1]
```diff
@@ -26,6 +26,9 @@
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Collections.Generic.ShortEnumEqualityComparer<>))]
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Collections.Generic.LongEnumEqualityComparer<>))]
 [assembly:System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Collections.ListDictionaryInternal))]
+#if netcoreapp
```
Comment (danmosemsft) [inline src/shims/manual/mscorlib.forwards.cs]: We need to remember to remove this (in master) when PN has the change, I guess we should have an issue

Comment (danmosemsft) [reply src/shims/manual/mscorlib.forwards.cs]: and the skip on the test

Comment (ViktorHofer) [reply src/shims/manual/mscorlib.forwards.cs]: This doesn't exist in master. I added these lines here for 2.0.

**Comment by safern** [comment]
@dotnet-bot test this please (CoreCLR updated)

**Comment by ViktorHofer** [comment]
Thanks Santi. I now first need to update the blobs.

**Comment by safern** [comment]
I just kicked off a new CoreFX 2.0.0 build. FYI: @weshaggard

**Pull request merged**